llvm/include/llvm/Analysis/DXILMetadataAnalysis.h

```diff
@@ -19,12 +20,16 @@ namespace llvm {
 
 namespace dxil {
 
+struct FunctionProperties {
+  unsigned NumThreads[3];
```

Comment:
We might need to put things in a union since not every shader profile has NumThreads.
Something like the PipelinePSVInfo union in PSV https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/BinaryFormat/DXContainer.h#L238

Reply:
> We might need to put things in a union since not every shader profile has NumThreads. Something like the PipelinePSVInfo union in PSV https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/BinaryFormat/DXContainer.h#L238

Wouldn't the `ShaderStage` field of `struct ModuleMetadataInfo` provide the necessary means for the consumer of the analysis info to identify and distinguish between shaders with functions with `numthreads` property/attribute? Creating a union as part of analysis information to represent the same information seems redundant.